### DISCUSSION TITLE
In generic code, if I want to return `something.asc()` or `something.desc()`, which type should I return?

### DISCUSSION THREAD

**Discussion by Zizico2** [diesel-rs/diesel#3483] [Q&A] [answered]
Returning an `Expression` with its `SqlType` being either `Asc<Something>` or `Desc<Something>` compiles, but it doesn't seem very intuitive since it can return either one. From digging into `Asc` and `Desc`, `NotSelectable` seems to be a good `SqlType`. Is there any other, intended way to do this?

**Comment by Zizico2** [comment, marked as answer]
This is my current return type: `Box<dyn Expression<SqlType = tankfinder_db_diesel_gen::diesel::expression::expression_types::NotSelectable>>`.

**Reply by weiznich** [reply]
Yes that's almost the correct solution. You might want to use [`BoxableExpresion`](https://docs.diesel.rs/2.0.x/diesel/expression/trait.BoxableExpression.html) instead of `Expression` there to make that code a bit more robust.

**Reply by Zizico2** [reply]
Thx
